api/src/main/java/net/kyori/adventure/text/JoinConfiguration.java

```diff
@@ -0,0 +1,300 @@
+ * being joined together is more than two. This can be used to insert a serial (or Oxford) comma if needed.</p>
+ *
+ * @see Component#join(JoinConfiguration, Iterable)
+ * @see Component#join(JoinConfiguration, ComponentLike...)
+ * @since 4.8.0
+ */
+@ApiStatus.NonExtendable
+public interface JoinConfiguration extends Buildable<JoinConfiguration, JoinConfiguration.Builder>, Examinable {
+  /**
+   * Creates a new builder.
+   *
+   * @return a new builder
+   * @since 4.8.0
+   */
+  static @NonNull Builder builder() {
+    return new JoinConfigurationImpl.BuilderImpl();
+  }
+
+  /**
+   * Gets a join configuration with no separators, prefix or suffix.
+   *
+   * @return the join configuration
+   * @since 4.8.0
+   */
+  static @NonNull JoinConfiguration noSeparators() {
+    return JoinConfigurationImpl.NULL;
+  }
+
+  /**
+   * Creates a join configuration with a separator and no prefix or suffix.
+   *
+   * @param separator the separator
+   * @return the join configuration
+   * @since 4.8.0
+   */
+  static @NonNull JoinConfiguration separator(final @Nullable ComponentLike separator) {
+    if(separator == null) return JoinConfigurationImpl.NULL;
+    return builder().separator(separator).build();
+  }
+
+  /**
+   * Creates a join configuration with a separator and last separator but no prefix or suffix.
+   *
+   * @param separator the separator
+   * @param lastSeparator the last separator
+   * @return the join configuration
+   * @since 4.8.0
+   */
+  static @NonNull JoinConfiguration separators(final @Nullable ComponentLike separator, final @Nullable ComponentLike lastSeparator) {
+    if(separator == null && lastSeparator == null) return JoinConfigurationImpl.NULL;
+    return builder().separator(separator).lastSeparator(lastSeparator).build();
+  }
+
+  /**
+   * Gets the prefix of this join configuration.
+   *
+   * @return the prefix
+   * @since 4.8.0
+   */
+  @Nullable Component prefix();
+
+  /**
+   * Gets the suffix of this join configuration.
+   *
+   * @return the suffix
+   * @since 4.8.0
+   */
+  @Nullable Component suffix();
+
+  /**
+   * Gets the separator of this join configuration.
+   *
+   * @return the separator
+   * @since 4.8.0
+   */
+  @Nullable Component separator();
+
+  /**
+   * Gets the last separator of this join configuration.
+   *
+   * @return the last separator
+   * @since 4.8.0
+   */
+  @Nullable Component lastSeparator();
+
+  /**
+   * Gets the operator of this join configuration.
+   *
+   * @return the operator
+   * @since 4.8.0
+   */
+  @NotNull UnaryOperator<Component> operator();
+
+  /**
+   * Gets the last separator that will be used instead of the normal last separator in the case where there
+   * are more than two components being joined. This can be used to mimic a serial (or Oxford) comma.
+   *
+   * @return the separator
+   * @since 4.8.0
+   */
+  @Nullable Component lastSeparatorIfSerial();
```

Comment:
Move above `operator` method.

Reply:
Speaking of, can anyone think of a better name for this? The current one is not very descriptive at all but the descriptive alternative `lastSeparatorIfThreeOrMoreComponentsAreBeingJoined()` is much, much worse.